client/signup/steps/user/index.jsx

```diff
@@ -732,8 +732,8 @@ export class UserStep extends Component {
 	}
 
 	render() {
-		if ( this.userCreationComplete() ) {
-			return null; // return nothing so that we don't see the completed signup form flash.
+		if ( this.userCreationComplete() && ! this.props.isWoo ) {
```

Comment:
The `userCreationComplete()` check was added in https://github.com/Automattic/wp-calypso/issues/91956. It causes the button to be re-enabled after the user logs in because when the signup flow is complete, there is also a logic that reset the signup controller in the parent controller and trigger a re-render. WPCOM flow need this because they have next steps but we don't. So we need to skip this.

Reply:
The other changes in this file is just refactor to replace `isWooOAuth2Client( this.props.oauth2Client )` with `this.props.isWoo`.